applications/TrilinosApplication/tests/cpp_tests/trilinos_cpp_test_utilities.cpp

```diff
@@ -234,14 +234,40 @@ void TrilinosCPPTestUtilities::CheckSparseMatrixFromLocalMatrix(
     values.reserve(total_size);
     for (std::size_t i = 0; i < rB.size1(); ++i) {
         for (std::size_t j = 0; j < rB.size2(); ++j) {
-            row_indexes.push_back(i);
-            column_indexes.push_back(j);
-            values.push_back(rB(i, j));
+            const double value = rB(i, j);
+            if (std::abs(value) > std::numeric_limits<double>::epsilon()) {
+                row_indexes.push_back(i);
+                column_indexes.push_back(j);
+                values.push_back(value);
+            }
         }
     }
     CheckSparseMatrix(rA, row_indexes, column_indexes, values);
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+void TrilinosCPPTestUtilities::CheckSparseMatrixFromLocalMatrix(
+    const TrilinosSparseMatrixType& rA,
+    const std::vector<int>& rRowIndexes,
+    const std::vector<int>& rColumnIndexes,
+    const TrilinosLocalMatrixType& rB,
```

Comment:
btw, in my tests i was normally using a map to store B.
why do you need it to be a dense matrix?